apps/api/src/app/widgets/dtos/mark-message-as-request.dto.ts

```diff
@@ -0,0 +1,31 @@
+import { IsBoolean, IsOptional, ValidateNested } from 'class-validator';
+import { ApiExtraModels, ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { ChannelPreference } from '../../shared/dtos/channel-preference';
+
+class MarkMessageFields {
+  @ApiProperty({
+    type: Boolean,
+    required: false,
+  })
+  seen?: boolean;
+
+  @ApiProperty({
+    type: Boolean,
+    required: false,
+  })
+  read?: boolean;
+}
+
+export class MarkMessageAsRequestDto {
+  @ApiProperty({
+    type: String,
+    isArray: true,
+    required: true,
+  })
```

Comment:
this will say that the type is a `string[]`, but actually it's a `string | string[]` 
Please change this to:
```
@ApiProperty({
    oneOf: [
      { type: 'string' },
      {
        type: 'array',
        items: {
          type: 'string',
        },
      },
    ],
    required: true,
  })
```

Reply:
Oh yes, I was thinking like this but couldn't figure out the exact properties to use.. can I get the docs to this @Api decorator for future reference?

Reply:
you can find it here: https://docs.nestjs.com/openapi/types-and-parameters#oneof-anyof-allof